packages/sdk-react/src/components/receivables/ReceivablesTable/ReceivablesTable.tsx

```diff
@@ -163,15 +258,37 @@ const ReceivablesTableBase = ({
 };
 
 /**
- * Manages the active tab state.
- * If the `tab` prop is provided, the component is controlled.
+ * Manages the active tab state for controlled mode and uncontrolled modes.
+ *
+ * - If the `tab === undefined || onTabChange === undefined`, the hook behaves uncontrolled hook;
+ *   that means it handles the internal state.
+ *   - Whenever `tab` is updated, the hook updates the internal state.
+ * - If the `tab !== undefined` and `onTabChange !== undefined`, the hook behaves controlled hook;
+ *   that means it just returns the original state.
  */
-const useSetActiveTab = ({
-  tab,
-  onTabChange,
+export const useControlledTab = ({
+  tab: controlledTab,
+  onTabChange: setControlledTab,
 }: Pick<ReceivablesTableProps, 'tab' | 'onTabChange'>) => {
-  const [tabControlled, onTabChangeControlled] =
-    useState<ReceivablesTableTabEnum>(ReceivablesTableTabEnum.Invoices);
+  const [uncontrolledTab, setUncontrolledTab] = useState(controlledTab ?? 0);
+
+  useEffect(() => {
+    if (controlledTab === undefined) return;
```

Comment:
Please refactor to `if (!conrolledTab) return`

Reply:
I refactored the logic completely as that doesn't seem to be needed and simplified it to a state and callback.